impala/_thrift_api.py

```diff
@@ -346,23 +382,34 @@ def get_http_transport(host, port, http_path, timeout=None, use_ssl=False,
         except AttributeError:
             auth = base64.encodestring(user_password).decode().strip('\n')
 
-        transport.setCustomHeaders({'Authorization': 'Basic %s' % auth})
+        def get_custom_headers(cookie_headers, has_auth_cookie):
+            custom_headers = {}
+            if cookie_headers:
+                log.debug('add cookies to HTTP header')
+                custom_headers['Cookie'] = cookie_headers
+            if not has_auth_cookie:
```

Comment:
Even though it is not strictly necessary, I would add the 'Authorization' header to the PLAIN/LDAP request even if the auth cookie is present, just to be on the safe side. I'm thinking about the unlikely event when at the request time the auth cookie is still valid, but by the time the server gets it, it has already expired.

The 'auth' value is created only once, so it won't cause a performance issue.

Reply:
Agree. Fixed it and added a comment.